### PR TITLE
State checking for object state updates

### LINKED ISSUE #608: Model and enforce valid state transitions in baseobject.py
This would be nice because it would be self documenting and catch errors when we introduce them.

### PR DESCRIPTION
**Pull request by mandoonandy**
Resolves #608

### CODE REVIEW

File: shakenfist/images.py [hunk 1]
```diff
@@ -68,7 +76,7 @@ def new(cls, url, checksum=None):
             'version': cls.current_version
         })
         i = Image.from_db(uuid)
-        i._db_set_attribute('state', {'state': 'initial'})
+        i._db_set_attribute('state', {'state': None})
         i.update_state('initial')
```
Comment (mikalstill) [inline shakenfist/images.py:80]: Why do this in two db writes? Do you see an advantage I'm missing?

**Review by mikalstill — COMMENTED** [review]
Just one quick question.

File: shakenfist/images.py (hunk 1 above)
Comment (mandoonandy) [reply shakenfist/images.py:80]: Minimal touching of the code. Just keeping this change to what was required. The change to that line was only the initial value.

I do agree that it is unnecessary. The related functions look like they will handle the missing DB entry.

Comment (mikalstill) [reply shakenfist/images.py:80]: Ok, I will merge this and then revisit.

**Comment by mikalstill** [comment]
Take a look at the CI results. There is definitely something up with this change.

**Comment by mandoonandy** [comment]
This has exposed a few issues... I will have to raise individual issues.

**Comment by mikalstill** [comment]
Should this one be a draft until the "other issues" are resolved? Or should I re-run CI?

**Comment by mandoonandy** [comment]
This did pass CI before my last comment. It should pass again...

**Comment by mandoonandy** [comment]
Failed due Internal Server Error in etcd during CI run.

**Review by mikalstill — APPROVED** [review]
LGTM

**Pull request merged**